### PR TITLE
[OYPD-532] GroupExPro demo content.

### PR DESCRIPTION
**Pull request by Kerby70**
Jira https://propeople-us.atlassian.net/browse/OYPD-532
D.o https://www.drupal.org/node/2893545

Make sure these boxes are checked before asking for review of your pull request - thank you!

## General checks
- [x] All coding styles are fulfilled and there are no any issues reported by CodeSniffer CI. <br/>CI code sniffer errors
- [x] All tests are running and there are no failed tests reported by CI. <br/>Behat test results
- [x] [Documentation](https://github.com/ymcatwincities/openy/tree/8.x-1.x/docs) has been updated according to PR changes.
- [x] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/>Steps for review
- [x] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md).
- [x] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/>drupal.org email
- [x] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

## Steps for review

- [x] See GroupEx Pro Demo content on
  - [x] 4 Demo branches
  - [x] on /schedules/group-schedules
  - [x] on /schedules/group-schedules-embedded
- [x] See menu items under schedules
  - [x] Schedules Framework to /schedules
  - [x] Group Schedules (Embedded) to /schedules/group-schedules-embedded
  - [x] Group Schedules to /schedules/group-schedules

### CODE REVIEW

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://ci.openymca.org/build1830 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build1830 |
| Installation process | http://install.openy-dev.ffwua.com/build1830/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.openymca.org:8080/job/PR_BUILDER/1830/

**Comment by gundevel** [comment]
Decoupling report http://ci.openymca.org/build1830

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://ci.openymca.org/build1830/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://ci.openymca.org/build1830/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://ci.openymca.org/build1830/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://ci.openymca.org/build1830/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://ci.openymca.org/build1830/blogWCAG2AAhtmlcs.txt

**Comment by gundevel** [comment]
Behat test results file URL: http://ci.openymca.org/build1830/profiles/contrib/openy/build/reports/behat

**Comment by AlexNetman** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://ci.openymca.org/build1850 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build1850 |
| Installation process | http://install.openy-dev.ffwua.com/build1850/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.openymca.org:8080/job/PR_BUILDER/1850/

**Comment by gundevel** [comment]
Decoupling report http://ci.openymca.org/build1850

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://ci.openymca.org/build1850/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://ci.openymca.org/build1850/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://ci.openymca.org/build1850/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://ci.openymca.org/build1850/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://ci.openymca.org/build1850/blogWCAG2AAhtmlcs.txt

**Comment by AlexNetman** [comment]
image
Like this:
image
That will result in displaying results for the needed branch on page load.

**Comment by gundevel** [comment]
Behat test results file URL: http://ci.openymca.org/build1850/profiles/contrib/openy/build/reports/behat

**Comment by Kerby70** [comment]
I will need to do further research to find how to set the block config in the demo content migration. It appears the data is stored in serialized format in paragraph__field_prgf_schedules_ref.field_prgf_schedules_ref_plugin_configuration the config variable that we need to set is enabled_locations. We have a MigrateProcessPlugin (openy_menu_icon) for menu icon data serialization that may point the way, we just need to ensure the other data is included & it won't be overwritten by those defaults.

**Comment by Kerby70** [comment]
We will follow-up on the branch configuration https://propeople-us.atlassian.net/browse/YGS-754

**Comment by david-hernandez** [comment]
When I try this I get the "block is broken or missing" error on the schedule page. The embedded one works. This is on a completely new install.

screen shot 2017-07-21 at 10 51 11 am

screen shot 2017-07-21 at 10 51 23 am

screen shot 2017-07-21 at 10 53 32 am

File: modules/openy_features/openy_demo_content/modules/openy_demo_nlanding/openy_demo_nlanding.install [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+/**
+ * @file OpenY Demon Landing page install file.
```
Comment (david-hernandez) [inline modules/openy_features/openy_demo_content/modules/openy_demo_nlanding/openy_demo_nlanding.install]: "Demon" typo here.

File: modules/openy_features/openy_demo_content/modules/openy_demo_nlanding/config/install/migrate_plus.migration.node_landing.yml [hunk 2]
```diff
@@ -137,6 +137,26 @@ source:
...
+      alias: '/schedules/group-schedules'
+      header_content_ids:
+        -
+          header_content_id: group_schedules
+      content_ids:
+        -
+          content_id: group_schedules_groupex_pro
+    -
+      id: group_schedules_embedded
+      title: 'Group Schedules(Embedded)'
```
Comment (david-hernandez) [inline modules/openy_features/openy_demo_content/modules/openy_demo_nlanding/config/install/migrate_plus.migration.node_landing.yml]: Add a space here between "Schedules" and "(Embedded)"

File: modules/openy_features/openy_demo_content/modules/openy_demo_menu/modules/openy_demo_menu_footer/config/install/migrate_plus.migration.menu_link_footer.yml [hunk 3]
```diff
@@ -147,6 +147,24 @@ source:
...
+      title: 'Group Schedules'
+      menu_name: 'footer-menu-center'
+      link: 'schedules/group-schedules'
+      weight: 1
+      parent_id: 15
+      expanded: 1
+      parent_link_path: 'schedules'
+    -
+      id: 17
+      title: 'Group Schedules(Embedded)'
```
Comment (david-hernandez) [inline modules/openy_features/openy_demo_content/modules/openy_demo_menu/modules/openy_demo_menu_footer/config/install/migrate_plus.migration.menu_link_footer.yml]: Add a space here between "Schedules" and "(Embedded)".

File: openy.profile [hunk 4]
```diff
@@ -269,6 +269,30 @@ function openy_discover_broken_paragraphs(array &$install_state) {
       $query->execute();
     }
   }
+
+  $schedule_paragraph_tables = ['paragraph__field_prgf_schedules_ref', 'paragraph_revision__field_prgf_schedules_ref'];
```
Comment (Kerby70) [inline openy.profile]: The code after this line should handle the broken paragraphs, though this is not new to this PR. I was not able to replicate that issue today.

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://ci.openymca.org/build1905 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build1905 |
| Installation process | http://install.openy-dev.ffwua.com/build1905/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.openymca.org:8080/job/PR_BUILDER/1905/

**Comment by gundevel** [comment]
Decoupling report http://ci.openymca.org/build1905

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://ci.openymca.org/build1905/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://ci.openymca.org/build1905/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://ci.openymca.org/build1905/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://ci.openymca.org/build1905/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://ci.openymca.org/build1905/blogWCAG2AAhtmlcs.txt

**Comment by gundevel** [comment]
Behat test results file URL: http://ci.openymca.org/build1905/profiles/contrib/openy/build/reports/behat

**Comment by Sanchiz** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://ci.openymca.org/build1916 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build1916 |
| Installation process | http://install.openy-dev.ffwua.com/build1916/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.openymca.org:8080/job/PR_BUILDER/1916/

**Comment by gundevel** [comment]
Decoupling report http://ci.openymca.org/build1916

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://ci.openymca.org/build1916/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://ci.openymca.org/build1916/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://ci.openymca.org/build1916/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://ci.openymca.org/build1916/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://ci.openymca.org/build1916/blogWCAG2AAhtmlcs.txt

**Comment by gundevel** [comment]
Behat test results file URL: http://ci.openymca.org/build1916/profiles/contrib/openy/build/reports/behat

File: openy.profile (hunk 4 above)
Comment (alexschedrov) [reply openy.profile]: That's correct. However this is code duplication. Can we move whole foreach to separate function that will accept:
- Condition field name, e.g. `field_prgf_schedules_ref_plugin_id`
- Serialization field name, e.g. `field_prgf_schedules_ref_plugin_configuration`
- any other params that are different...
- Array with tables.

**Review by alexschedrov — CHANGES_REQUESTED** [review]

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://ci.openymca.org/build1923 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build1923 |
| Installation process | http://install.openy-dev.ffwua.com/build1923/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.openymca.org:8080/job/PR_BUILDER/1923/

**Comment by gundevel** [comment]
Decoupling report http://ci.openymca.org/build1923

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://ci.openymca.org/build1923/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://ci.openymca.org/build1923/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://ci.openymca.org/build1923/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://ci.openymca.org/build1923/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://ci.openymca.org/build1923/blogWCAG2AAhtmlcs.txt

**Comment by gundevel** [comment]
Behat test results file URL: http://ci.openymca.org/build1923/profiles/contrib/openy/build/reports/behat

**Comment by Sanchiz** [comment]
retest this please

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://ci.openymca.org/build1926 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build1926 |
| Installation process | http://install.openy-dev.ffwua.com/build1926/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.openymca.org:8080/job/PR_BUILDER/1926/

**Comment by bjwilkins** [comment]
Decoupling report http://ci.openymca.org/build1926

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://ci.openymca.org/build1926/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://ci.openymca.org/build1926/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://ci.openymca.org/build1926/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://ci.openymca.org/build1926/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://ci.openymca.org/build1926/blogWCAG2AAhtmlcs.txt

**Comment by bjwilkins** [comment]
Behat test results file URL: http://ci.openymca.org/build1926/profiles/contrib/openy/build/reports/behat

**Review by alexschedrov — APPROVED** [review]

**Pull request merged**